gluoncv/model_zoo/hrnet.py

```diff
@@ -0,0 +1,593 @@
+    (3,    'BASIC',      [4]*4,         [44,88,176,352],'SUM')
+]
+
+hrnet_spec['w48'] = [
+    #modules, block_type, blocks, channels, fuse_method
+    (1,    'BOTTLENECK', [4],           [64],           'SUM'),
+    (1,    'BASIC',      [4]*2,         [48,96],        'SUM'),
+    (4,    'BASIC',      [4]*3,         [48,96,192],    'SUM'),
+    (3,    'BASIC',      [4]*4,         [48,96,192,384],'SUM')
+]
+
+def get_hrnet(model_name, stage_interp_type='nearest', purpose='cls', pretrained=False, ctx=cpu(),
+                 root='~/.mxnet/models', norm_layer=BatchNorm, norm_kwargs=None, num_classes=1000, **kwargs):
+    r"""MobileNet model from the
+    `"Searching for MobileNetV3"
+    <https://arxiv.org/abs/1905.02244>`_ paper.
+
+    Parameters
+    ----------
+    model_name : string
+        The name of hrnet models: w18_small_v1/w18_small_v2/w30/w32/w40/w42/w48.
+    stage_interp_type : string
+        The interpolation type for upsample in each stage, nearest, bilinear and
+        bilinear_like are supported.
+    purpose: string
+        The purpose of model, cls and seg are supported.
+    pretrained : bool or str
+        Boolean value controls whether to load the default pretrained weights for model.
+        String value represents the hashtag for a certain version of pretrained weights.
+    ctx : Context, default CPU
+        The context in which to load the pretrained weights.
+    root : str, default $MXNET_HOME/models
+        Location for keeping the model parameters.
+    norm_layer : object
+        Normalization layer used (default: :class:`mxnet.gluon.nn.BatchNorm`)
+        Can be :class:`mxnet.gluon.nn.BatchNorm` or :class:`mxnet.gluon.contrib.nn.SyncBatchNorm`.
+    norm_kwargs : dict
+        Additional `norm_layer` arguments, for example `num_devices=4`
+        for :class:`mxnet.gluon.contrib.nn.SyncBatchNorm`.
+    """
+    if model_name not in hrnet_spec.keys():
+        raise NotImplementedError
+
+    spec = hrnet_spec[model_name]
+
+    if purpose == 'cls':
+        net = HighResolutionClsNet(spec, stage_interp_type, norm_layer,
+            norm_kwargs, num_classes, **kwargs)
+    elif purpose == 'seg':
+        net = HighResolutionSegNet(spec, stage_interp_type, norm_layer,
+            norm_kwargs, num_classes, **kwargs)
+    else:
+        raise NotImplementedError
+
+    if pretrained:
+        raise RuntimeError('Pretrained model of hrnet is not available now.')
+    
+    return net
+
+def hrnet_w18_small_v1_c(**kwargs):
+    return get_hrnet('w18_small_v1', **kwargs)
+
+def hrnet_w18_small_v2_c(**kwargs):
+    return get_hrnet('w18_small_v2', **kwargs)
+
+def hrnet_w30_c(**kwargs):
+    return get_hrnet('w30', **kwargs)
+
+def hrnet_w32_c(**kwargs):
+    return get_hrnet('w32', **kwargs)
+
+def hrnet_w40_c(**kwargs):
+    return get_hrnet('w40', **kwargs)
+
+def hrnet_w44_c(**kwargs):
+    return get_hrnet('w44', **kwargs)
+
+def hrnet_w48_c(**kwargs):
```

Comment:
so, I suppose you are trying to use `c` or `s` to separate classification or segmentation models right?
It's very confusing from the name itself, maybe we need better naming for them

Reply:
I looked at paddlepaddle's name scheme, I think I can add some comment. Do you have better idea?